### PR TITLE
[MOB-21377] added support for debug events generated by external systems

### PR DESCRIPTION
**Pull request by akhiljain1907**
## Description

This PR add support of Debug Events triggered by external system. Also, solves issue - [MOB-21377](https://jira.corp.adobe.com/browse/MOB-21377)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by spoorthipujariadobe — COMMENTED** [review]
The logic here looks good. Just a question on how customers need to use simulate workflow. You're dispatching Optimize Notification event which triggers `onPropositionsUpdate` API callback to be called. Is our recommendation that customers implement this API for preview purpose? Would you be able to add a page in the Github docs on what changes they need to make for simulate if they are already using updatePropostions + getPropositions in their app?

**Comment by akhiljain1907** [comment]
> The logic here looks good. Just a question on how customers need to use simulate workflow. You're dispatching Optimize Notification event which triggers `onPropositionsUpdate` API callback to be called. Is our recommendation that customers implement this API for simulate purpose? Would you be able to add a page in the Github docs on what changes they need to make for simulate if they are already using updatePropostions + getPropositions in their app?

As in case of simulate we don't want to corrupt the existing cache, we need to provide an api to customers where they should listen to such type of Debug Events triggered by external system. Currently we have this onPropositionsUpdate api in our code and so i used the same. Do you think should add a new api like 'onSimulate' ? Or is it okay to use onPropositionUpdate ?

File: Sources/AEPOptimize/Optimize.swift [hunk 1]
```diff
@@ -502,6 +507,45 @@ public class Optimize: NSObject, Extension {
...
+    /// Processes debug events triggered by the system.
+    ///
+    /// A debug event allows the optimize extension to processes non-production workflows.
+    /// - Parameter event: the debug `Event` to be handled.
+    private func processDebugEvent(_ event: Event) {
+        guard event.isDebugEvent else {
+            Log.debug(label: OptimizeConstants.LOG_TAG,
+                      " Ignoring Debug event, either handle type is not com.adobe.eventType.system or source is not com.adobe.eventSource.debug")
+            return
+        }
```
Comment (sbenedicadb) [inline Sources/AEPOptimize/Optimize.swift:519]: would it be better here to validate that our `debug.eventType` and `debug.eventSource` values exist and are something we want to handle?

Comment (akhiljain1907) [reply Sources/AEPOptimize/Optimize.swift:519]: Currently I am not receiving debug object in event's data.
Here is an example -
{
  "id": "D2C4E6D7-EF0A-406F-8460-DBD179EFA7D3",
  "name": "AEP Response Event Handle (Spoof)",
  "type": "com.adobe.eventType.system",
  "source": "com.adobe.eventSource.debug",
  "data": {
    "type": "personalization:decisions",
    "payload": [
      {
        "scopeDetails": {
          "decisionProvider": "TGT",
          "activity": {
            "id": "1451350"
          },
          "experience": {
            "id": "1"
          },
          "correlationID": "1451350:1:0",
          "strategies": [
            {
              "step": "entry",
              "trafficType": "0"
            },
            {
              "trafficType": "0",
              "step": "display"
            },
            {
              "trafficType": "0",
              "step": "custom-metric-3"
            },
            {
              "step": "conversion",
              "trafficType": "0"
            }
          ],
          "characteristics": {
            "eventToken": "ozJDe5qxmh1ynJiKM3qS7w6taum/Ule9G8NTrZHPdMm7I2aV5okLvdambMy5fCYKvxeaHLn+wzXtAu/7/KZ4kA=="
          }
        },
        "id": "AT:eyJhY3Rpdml0eUlkIjoiMTQ1MTM1MCIsImV4cGVyaWVuY2VJZCI6IjEifQ==",
        "scope": "optimize-tutorial-loc",
        "items": [
          {
            "data": {
              "content": "<html><body></body></html>",
              "id": "0",
              "format": "text/html"
            },
            "id": "0",
            "schema": "https://ns.adobe.com/personalization/html-content-item",
            "meta": {
              "profile.favoriteColor": "black",
              "profile.Happy_PS": "[hello Happy Profile Script, , , optimize-tutorial-loc, , landing url is, ]",
              "profile.daysSinceLastVisit": "5.0",
              "activity.name": "Optimize Tutorial Activity",
              "profile.marketingCloudVisitorId": "12797319901314597948242657659519290950",
              "profile.impressions": "40",
              "experience.name": "Experience A",
              "profile.greetingsAffinity": "[Good afternoon, Good evening]",
              "profile.tntId": "12797319901314597948242657659519290950-lKYpJH"
            }
          }
        ]
      }
    ]
  },
  "timestamp": "2024-11-19 03:13:26 +0000",
  "responseId": "nil",
  "parentId": "nil",
  "mask": "nil"
}

**Comment by akhiljain1907** [comment]
Raised another PR with debug event support and separate preview cache and thus closing this one.

**Pull request closed without merging**